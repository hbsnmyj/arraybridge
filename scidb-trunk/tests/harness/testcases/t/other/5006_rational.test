--setup
--start-query-logging
# Ticket number: 5006
# A simpler repro for the error as revealed in other.binary_load9_udt_aql.

load_library('rational')
load_library('point')
create array A <a:int32>[x=0:0,1,0]
create array Y <a:int32,c:rational> [x=0:0,1,0]
store(build(A,x+1),A)
store(apply(A,c,point2str(str2point('(3,2)'))),ZZ)

--test

save(ZZ, 'mat.dmp', -2, '(int32,string)')
load(Y, 'mat.dmp', -2, '(int32,string)', 100, Y_shad)

--cleanup
remove(A)
remove(Y)
remove(ZZ)
remove(Y_shad)

