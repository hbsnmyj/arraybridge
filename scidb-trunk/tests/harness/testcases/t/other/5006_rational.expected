SCIDB QUERY : <load_library('rational')>
Query was executed successfully

SCIDB QUERY : <load_library('point')>
Query was executed successfully

SCIDB QUERY : <create array A <a:int32>[x=0:0,1,0]>
Query was executed successfully

SCIDB QUERY : <create array Y <a:int32,c:rational> [x=0:0,1,0]>
Query was executed successfully

SCIDB QUERY : <store(build(A,x+1),A)>
{x} a
{0} 1

SCIDB QUERY : <store(apply(A,c,point2str(str2point('(3,2)'))),ZZ)>
{x} a,c
{0} 1,'(3,2)'

SCIDB QUERY : <save(ZZ, 'mat.dmp', -2, '(int32,string)')>
{x} a,c

SCIDB QUERY : <load(Y, 'mat.dmp', -2, '(int32,string)', 100, Y_shad)>
{x} a,c
{0} 1,null

SCIDB QUERY : <remove(A)>
Query was executed successfully

SCIDB QUERY : <remove(Y)>
Query was executed successfully

SCIDB QUERY : <remove(ZZ)>
Query was executed successfully

SCIDB QUERY : <remove(Y_shad)>
Query was executed successfully

